src/blr/cli/tagfastq.py

```diff
@@ -119,22 +101,44 @@ def parse_corrected_barcodes(open_file):
     return corrected_barcodes
 
 
-def parse_raw_barcodes(open_file):
-    """
-    Parse FASTA/FASTQ containing barcode sequences and return a dictionary with entry headers
-    pointing to a raw barcodes sequence.
-    :param open_file: dnaio odject.
-    :return: dict: entry headers pointing to a raw barcodes sequence
-    """
-    for barcode in tqdm(open_file, desc="Raw barcodes processed"):
-        header, _ = barcode.name.split(maxsplit=1)
-        yield {header: barcode.sequence}
+class BarcodeReader:
+    def __init__(self, filename):
+        self._cache = dict()
+        self._file = dnaio.open(filename, mode="r")
+        self.barcodes = self.parse()
+
+    def parse(self):
+        for barcode in tqdm(self._file, desc="Uncorrected barcodes processed"):
+            header, _ = barcode.name.split(maxsplit=1)
+            yield {header: barcode.sequence}
```

Comment:
I suggest to call it `read_name` or perhaps `read_id` instead of `header` ("header" would imply for me that it includes the entire line, even including the `@`).